### PR TITLE
Add proto files to support the value this service will store

### PR DESCRIPTION
**Pull request by schancel**
Summary:
In order to validate the signature properly, we must have a canonical way to
serialize map data. Therefore, we are switching to Protobufs from JSON as the
canonical way to serialize data.  While this requires additional dependencies,
protobufs was already needed for BIP70 support.  This is a good tradeoff to make
since JSON fields are unordered.

Test:
  go test ./...

### CODE REVIEW

File: proto/addressmetadata.proto [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+message Header {
+    string Name = 1;
+    string Value = 2;
+}
+
+// Row is an indidual piece of structured data provided by wallet authors.
+// The headers help wallets determine how to deserialize the data for their purposes.
+message MetadataField {
+	repeated Header Headers = 1;
```
Comment (FrankFusion905) [inline proto/addressmetadata.proto]: Do we even want to separate the headers as opposed to just storing everything as some arbitrary bytes?

File: proto/addressmetadata.proto [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	// Serialized version of the XPubKey.  The *hash* of this XPub should correspond to the `key` in the kv store
+	bytes PubKey = 1;
+	// Signature is the signature of the metadata by XPubKey
+	bytes Signature = 2;
+	// Signature type provided.  Default is Schnorr, but can be ecdsa
+	enum SignatureType {
+        Schnorr = 0;
+        ECDSA = 1;
+    }
+    SignatureType Type = 3;
```
Comment (FrankFusion905) [inline proto/addressmetadata.proto]: Do we want a default value?

File: proto/addressmetadata.proto (hunk 1 above)
Comment (schancel) [reply proto/addressmetadata.proto]: I think we should define it here so the downstream consumers can at least guaranteee there's some metadata they can parse before deciding what to do with the actual Values.

The alternative is to have a "by convention" way to specify what type of data is in the payload anyways.

File: proto/addressmetadata.proto (hunk 2 above)
Comment (schancel) [reply proto/addressmetadata.proto]: Schnorr is the default right?

Comment (FrankFusion905) [reply proto/addressmetadata.proto]: I've always done `SignatureType Type = 3 [default = Schnorr];`

Comment (schancel) [reply proto/addressmetadata.proto]: Okay, Will fix

Comment (schancel) [reply proto/addressmetadata.proto]:
```
shammah @ king-george.local ➜  proto git:(proto-update) ✗ protoc --go_out=../pkg/models/ *.proto
addressmetadata.proto: Explicit default values are not allowed in proto3.
```

**Pull request merged**